### DISCUSSION TITLE
Where to find implementation of I2C.scan() class?

### DISCUSSION THREAD

**Discussion by AL-M0ND** [micropython/micropython#15344] [RP2040 / Pico] [answered]
Basically what the title says.

I found the machine_i2c.c class (ports/rp2/machine_i2c.c), but I'm trying to understand how the I2C.scan() method works and it doesn't look like that method is implemented there.
Is the source for it available here (or anywhere?)

**Comment by miketeachman** [comment, marked as answer]
It's located in a second file having the same name, `machine_i2c.c`, in the directory containing all the cross-platform code called `extmod`.  The scan method is  [here](https://github.com/micropython/micropython/blob/908ab1ceca15ee6fd0ef82ca4cba770a3ec41894/extmod/machine_i2c.c#L322).

**Reply by AL-M0ND** [reply]
Thanks so much!
